x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityCacheRegistry.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.security.support;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.elasticsearch.xpack.security.support.SecurityIndexManager.isIndexDeleted;
+import static org.elasticsearch.xpack.security.support.SecurityIndexManager.isMoveFromRedToNonRed;
+
+public class SecurityCacheRegistry {
+
+    private static final Map<String, CacheInvalidator> cacheInvalidators = new ConcurrentHashMap<>();
```

Comment:
I think we probably prefer to have a singleton instance for this class similar to all other components. If so, I can make it so and have it injected into the Transport action.